src/modules/access/access.service.ts

```diff
@@ -128,250 +90,141 @@ type Role = 'admin' | 'underboss' | 'caporegime' | 'host' | null;
  */
 @Injectable()
 export class AccessService {
-  constructor(private readonly knexService: KnexService) {}
+  constructor(private readonly knexService: KnexService, private readonly regionService: RegionService, private readonly countryService: CountryService, private readonly cityService: CityService) {}
 
-  /**
-   * Retrieves all regions from the database
-   * @returns {Promise<Array<{id: string, name: string}>>} Array of regions with their IDs and names
-   */
-  async getAllRegions(): Promise<{ id: string; name: string }[]> {
-    return this.knexService.knex('region').select('id', 'name');
+  async getRegionAccess(telegram_id: string): Promise<IRegionAccess[]> {
+    return this.knexService.knex('region_access').where('user_telegram_id', telegram_id);
   }
 
-  /**
-   * Retrieves a specific region by its ID
-   * @param {string} regionId - The unique identifier of the region
-   * @returns {Promise<{id: string, name: string} | undefined>} Region data or undefined if not found
-   */
-  async getRegionById(regionId: string): Promise<{ id: string; name: string } | undefined> {
-    return this.knexService.knex('region').where('id', regionId).first();
+  async getCountryAccess(telegram_id: string): Promise<ICountryAccess[]> {
+    return this.knexService.knex('country_access').where('user_telegram_id', telegram_id);
   }
 
-  /**
-   * Gets all cities belonging to a specific region
-   * @param {string} regionId - The unique identifier of the region
-   * @returns {Promise<CityData[]>} Array of cities with their details
-   */
-  async getCitiesByRegion(regionId: string): Promise<CityData[]> {
-    return this.knexService
-      .knex('city')
-      .join('country', 'city.country_id', 'country.id')
-      .where('country.region_id', regionId)
-      .select(
-        'city.id as city_id',
-        'city.name as city_name',
-        'city.group_id',
-        'city.telegram_link',
-      );
+  async getCityAccess(telegram_id: string): Promise<ICityAccess[]> {
+    return this.knexService.knex('city_access').where('user_telegram_id', telegram_id);
   }
 
   /**
-   * Gets the role of a user based on their Telegram ID
+   * Gets detailed access information for a user
    * @param {string} telegram_id - The Telegram ID of the user
-   * @returns {Promise<Role>} The user's role or null if no role is assigned
+   * @returns {Promise<AccessResult>} Detailed access information including cities, regions, and countries
    */
-  async getAccessRole(telegram_id: string): Promise<Role> {
+  async getUserAccess(telegram_id: string): Promise<TAccessResult | null> {
```

Comment:
I've updated the implementation of getUserAccess method here. Broadcast needs to be updated accordingly.